flink-python/pyflink/table/tests/test_descriptor.py

```diff
@@ -0,0 +1,607 @@
+            "sink",
+            field_names, field_types, CsvTableSink(sink_path))
+
+        # connect source
+        t_env.connect(FileSystem().path(source_path))\
+             .with_format(OldCsv()
+                          .field_delimiter(',')
+                          .field("a", DataTypes.INT)
+                          .field("b", DataTypes.STRING)
+                          .field("c", DataTypes.STRING))\
+             .with_schema(Schema()
+                          .field("a", DataTypes.INT)
+                          .field("b", DataTypes.STRING)
+                          .field("c", DataTypes.STRING))\
+             .register_table_source("source")
+        t_env.scan("source") \
+             .select("a + 1, b, c") \
+             .insert_into("sink")
+        t_env.execute()
+
+        with open(sink_path, 'r') as f:
+            lines = f.read()
+            assert lines == '2,Hi,Hello\n' + '3,Hello,Hello\n'
+
+    def test_register_table_source_and_sink(self):
+        source_path = os.path.join(self.tempdir + '/streaming.csv')
+        field_names = ["a", "b", "c"]
+        field_types = [DataTypes.INT, DataTypes.STRING, DataTypes.STRING]
+        data = [(1, "Hi", "Hello"), (2, "Hello", "Hello")]
+        self.prepare_csv_source(source_path, data, field_types, field_names)
+        sink_path = os.path.join(self.tempdir + '/streaming2.csv')
+        if os.path.isfile(sink_path):
+            os.remove(sink_path)
+        t_env = self.t_env
+
+        t_env.connect(FileSystem().path(source_path))\
+             .with_format(OldCsv()
+                          .field_delimiter(',')
+                          .field("a", DataTypes.INT)
+                          .field("b", DataTypes.STRING)
+                          .field("c", DataTypes.STRING))\
+             .with_schema(Schema()
+                          .field("a", DataTypes.INT)
+                          .field("b", DataTypes.STRING)
+                          .field("c", DataTypes.STRING))\
+             .register_table_source_and_sink("source")
+        t_env.connect(FileSystem().path(sink_path))\
+             .with_format(OldCsv()
+                          .field_delimiter(',')
+                          .field("a", DataTypes.INT)
+                          .field("b", DataTypes.STRING)
+                          .field("c", DataTypes.STRING))\
+             .with_schema(Schema()
+                          .field("a", DataTypes.INT)
+                          .field("b", DataTypes.STRING)
+                          .field("c", DataTypes.STRING))\
+             .register_table_source_and_sink("sink")
+        t_env.scan("source") \
+             .select("a + 1, b, c") \
+             .insert_into("sink")
+        t_env.execute()
+
+        with open(sink_path, 'r') as f:
+            lines = f.read()
+            assert lines == '2,Hi,Hello\n' + "3,Hello,Hello\n"
+
+
+class StreamTableDescriptorTests(PyFlinkStreamTableTestCase, AbstractTableDescriptorTests):
+
```

Comment:
Please add `toProperties()`test case?

Reply:
In fact most test cases for StreamTableDescriptor use to_properties to get the test results. So Maybe an independent to_properties test cases is not necessary?

Reply:
Hmm, I think keeping the current test case is enough. ;)